languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -11906,9 +11881,193 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
                 <example>Torna-se necessário obedecer a normas de conduta.</example>
                 <example>A Literatura obedece a leis inflexíveis: a da herança, a do meio, a do momento.</example>
             </rule>
-        </rulegroup>
+            <!-- Moved from the PHRASAL_VERBS_ERRORS rulegroup above and split to handle contractions. -->
+            <rule>
+                <antipattern>
+                    <token min='0' regexp='yes'>me|te|se|vos</token>
+                    <token inflected='yes'>habituar</token>
+                    <token min='0' regexp='yes' spacebefore='no'>&hifen;</token>
+                    <token min='0' regexp='yes'>me|te|se|nos|vos</token>
+                    <token>com</token>
+                    <token regexp="yes">a(quel[ea])?s?|os?</token>
+                </antipattern>
+                <antipattern>
+                    <token min='0' regexp='yes'>me|te|se|vos</token>
+                    <token inflected='yes'>habituar</token>
+                    <token min='0' regexp='yes' spacebefore='no'>&hifen;</token>
+                    <token min='0' regexp='yes'>me|te|se|nos|vos</token>
+                    <token regexp="yes">co(m|n[st])igo|conv?osco</token>
+                </antipattern>
+                <pattern>
+                    <token min='0' regexp='yes'>me|te|se|vos</token>
+                    <token inflected='yes'>habituar</token>
+                    <token min='0' regexp='yes' spacebefore='no'>&hifen;</token>
+                    <token min='0' regexp='yes'>me|te|se|nos|vos</token>
+                    <marker>
+                        <token>com</token>
+                    </marker>
+                </pattern>
+                <message>Neste contexto, o verbo &quot;habituar&quot; rege-se com a preposição &quot;a&quot;.</message>
+                <suggestion>a</suggestion>
+                <example correction='a'>Ainda não me habituei <marker>com</marker> ele.</example>
+                <example correction='a'>Tenho dificuldades em habituar-me <marker>com</marker> novas rotinas.</example>
+                <example>Consegues habituar-te com calma ao clima.</example>
+            </rule>
+            <rule>
+                <pattern>
+                    <token min='0' regexp='yes'>me|te|se|vos</token>
```

Comment:
I think the first `<token min='0' regexp='yes'>me|te|se|vos</token>` in the rules are useless. If min="0" is active, then it doesn't matter what comes before the verb "habituar". We could remove this line altogether and test it like this, or we could create further subrules to filter cases including pronouns.

Reply:
Fair – these were there from the previous version of the rule, which, admittedly, repeated the whole structure in the suggestion.